### PR TITLE
Scaffold frontpage

### PR DESCRIPTION
**Pull request by mikeyrayvon**

### CODE REVIEW

File: front-page.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    <?php get_template_part('partials/programacion'); ?>
+  </section>
+
+<?php
+  $page_sobre = get_page_by_path('sobre');
+
+  if ($page_sobre != null) {
+?>
+  <section id="page-sobre" class="page-overlay">
+    <?php echo apply_filters('the_content', $page_sobre->post_content); ?>
```
Comment (varffs) [inline front-page.php]: Would it be nicer to setup post data here so you can just use `the_content()` and maybe in the future use other template tags? I'm not sure what is best practice

File: front-page.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  </section>
+
+  <section id="page-programacion" class="page-overlay">
+    <?php get_template_part('partials/programacion'); ?>
+  </section>
+
+<?php
+  $page_sobre = get_page_by_path('sobre');
+
+  if ($page_sobre != null) {
```
Comment (varffs) [inline front-page.php]: the empty return value of `get_page_by_path` is null which is false as a condition. so the check can just be on the variable (99% sure of this)

File: lib/functions-custom.php [hunk 3]
```diff
@@ -1,3 +1,39 @@
...
+function get_events_array( $query_args = '' ) {
+  $args = wp_parse_args( $query_args, array(
+    'post_type' => 'evento',
+    'meta_key' => '_igv_evento_start',
+    'order' => 'ASC',
+    'orderby' => 'meta_value',
+    'meta_query' => array(
+      array(
+        'key'     => '_igv_evento_start',
+        'value'   => current_time('timestamp') - (60 * 60 * 24), // UTC -5, 6 hours ago
```
Comment (varffs) [inline lib/functions-custom.php]: moment.php is built in to our boiler and will generate these time values with a much more readable codebase. There is a function `subtractHours()` on moment objects that will do this manipulation readably

However why does the function subtract 6 hours? So there is stuff that previously happened, or is this related to servertime?

File: lib/functions-custom.php [hunk 4]
```diff
@@ -1,3 +1,39 @@
...
+      )
+    )
+  ) );
+  $posts = get_posts( $args );
+  $event_array = array();
+
+  if ( $posts ) {
+    foreach ( $posts as $post ) {
+      $timestamp = get_post_meta($post->ID, '_igv_evento_start', true);
+      $year = date('Y', $timestamp);
```
Comment (varffs) [inline lib/functions-custom.php]: This is obv more readable that previous context, but again moment.php will be much nicer to write and read when getting these values.

File: lib/functions-custom.php [hunk 5]
```diff
@@ -1,3 +1,39 @@
 <?php
 
-// Custom functions (like special queries, etc)
\ No newline at end of file
+// Custom functions (like special queries, etc)
+
+function get_events_array( $query_args = '' ) {
```
Comment (varffs) [inline lib/functions-custom.php]: A comment to outline what this function does would be very helpful

**Review by varffs — CHANGES_REQUESTED** [review]
When we manipulate time in php I 100% think we should be using the moment.php library https://github.com/fightbulc/moment.php (and the js one when we do in js). It is prebuilt into the boiler and allows much more readable code than any other way of working with time.

File: lib/functions-custom.php (hunk 3 above)
Comment (mikeyrayvon) [reply lib/functions-custom.php]: ah it actually subtracts 24 hours. forgot to update the comment. yea its to show what already happened

**Comment by mikeyrayvon** [comment]
moment is only useful when formatting datetime to text. in this case im still manipulating/passing unix timestamps so moment doesnt come into play. were going to organize the schedule dom/content on the front end in js so will use moment there.

**Comment by varffs** [comment]
moment.php outputs timestamps with `->format('U')` and we've used moment.php to manipulate dates for queries on all the last projects that have used timestamps datepickers. I've written most of that codebase I guess and I promise moment.php makes doing so easier

**Comment by mikeyrayvon** [comment]
in this case it was easier to do it by hand.

most of the manipulation is going on in js, and were using moment.js there.

**Pull request merged**